File: day7/part1.py
from os import path
from statistics import median

with open(path.join(path.dirname(__file__), 'input.txt'), 'r') as file:
    input = list(map(int, file.read().split(',')))

med = int(median(input))
result = sum(abs(med - val) for val in input)

print(result)

File: day02/part1.js
const fs = require('fs');
const path = require('path');

const input = fs.readFileSync(path.join(__dirname, 'input.txt')).toString().trim().split('\n');

let horizontal = 0;
let depth = 0;

for (const command of input) {
    let [, dir, dist] = command.match(/([a-z]+) ([0-9]+)/);
    dist = Number.parseInt(dist);

    if (dir === 'forward') {
        horizontal += dist;
    } else if (dir === 'down') {
        depth += dist;
    } else if (dir === 'up') {
        depth -= dist;
    } else {
        throw Error();
    }
}

console.log(horizontal * depth);
